internal/store/driver.go

```diff
@@ -522,26 +525,59 @@ func (d *Driver) calculateIngressLoadBalancerIPStatus(ing *netv1.Ingress, c clie
 	return status
 }
 
-func (d *Driver) getBackendServicePort(backendSvc netv1.IngressServiceBackend, namespace string) (int32, error) {
-	servicePort := backendSvc.Port.Number
-	if servicePort > 0 {
-		return servicePort, nil
+func (d *Driver) getBackendServicePort(backendSvc netv1.IngressServiceBackend, namespace string) (int32, string, error) {
+	service, err := d.GetServiceV1(backendSvc.Name, namespace)
+	if err != nil {
+		return 0, "", err
 	}
 
-	d.log.V(3).Info("Searching service for named port", "namespace", namespace, "service", backendSvc.Name, "port.name", backendSvc.Port.Name)
-	service, err := d.GetServiceV1(backendSvc.Name, namespace)
+	servicePort, err := d.findServicesPort(service, backendSvc.Port)
+	if err != nil {
+		return 0, "", err
+	}
+
+	protocol, err := d.getPortAnnotatedProtocol(service, servicePort.Name)
 	if err != nil {
-		return 0, err
+		return 0, "", err
 	}
 
+	return servicePort.Port, protocol, nil
+}
+
+func (d *Driver) findServicesPort(service *corev1.Service, backendSvcPort netv1.ServiceBackendPort) (*corev1.ServicePort, error) {
 	for _, port := range service.Spec.Ports {
-		if port.Name == backendSvc.Port.Name {
-			d.log.V(3).Info("Found named port for service", "namespace", namespace, "service", service.Name, "port.name", port.Name, "port.number", port.Port)
-			return port.Port, nil
+		if (backendSvcPort.Number > 0 && port.Port == backendSvcPort.Number) || port.Name == backendSvcPort.Name {
+			d.log.V(3).Info("Found matching port for service", service.Name, "namespace", service.Namespace, "port.name", port.Name, "port.number", port.Port)
```

Comment:
```suggestion
                        d.log.V(3).Info("Found matching port", "service", service.Name, "namespace", service.Namespace, "port.name", port.Name, "port.number", port.Port)
```
The logger expects an even number of arguments after the message otherwise it will ignore the last key.
```
2023-06-01T05:54:32Z	LEVEL(-3)	cache-store-driver	Found matching port for service	{"blog": "namespace", "default": "port.name", "": "port.number"}
2023-06-01T05:54:32Z	DPANIC	cache-store-driver	odd number of arguments passed as key-value pairs for logging	{"ignored key": 443}

```